crates/wdk-build/src/lib.rs

```diff
@@ -755,6 +755,21 @@ impl Config {
             headers.extend(["hidpddi.h", "hidport.h", "kbdmou.h", "ntdd8042.h"]);
         }
 
+        if matches!(self.driver_config, DriverConfig::Umdf(_)) {
+            /* the hidport.h file is expected for Umdf, but located in km folder */
+            let include_directory = self.wdk_content_root.join("Include");
+            let sdk_version = utils::get_latest_windows_sdk_version(include_directory.as_path()).expect("sdk version");
+            let windows_sdk_include_path = include_directory.join(sdk_version);
+            let hidports_path = windows_sdk_include_path
+                .join("km")
+                .join("hidport.h")
+                .canonicalize().unwrap()
+                .strip_extended_length_path_prefix().unwrap()
+                .to_string_lossy().to_string().replace("\\", "/");
+
+            headers.push(Box::leak(hidports_path.into_boxed_str()));
```

Comment:
This is rather ugly. But my rust-fy is oh so limited. These functions likely should return some Cow string or just plain String instead of some static reference, but that feel above my skill lever light now.

Reply:
I agree that this is ugly. we should not leak memory here. 

Here's a simple example of how Cow would work here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=201ce300262dabda495bc0b36b2bc3e0

You can change the signature of this private function to return Cow<'static, str>

Reply:
Wasn't soo simple due to side effects of the match in headers(). Can you help me out here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=0e50c887aa9858043dbfbdb196c354d0 

I can't understand how it can be this hard to get this match statement to end up with compatible types. All the into, into_iters, to_owned is breaking my mind at the moment.

Reply:
Ok. I think i got some solution to this.

Ps. Rust strings are really hard to work with even for somebody with lots of c and c++ background. Feels like so much premature optimization. But maybe I'll learn at some point :).